src/main/java/hudson/plugins/mercurial/HgExe.java

```diff
@@ -346,6 +346,19 @@ public ProcStarter run(ArgumentListBuilder args) {
     }
 
     /**
+     * Gets the branch name of given revision number or of the current workspace.
+     * @param rev the revision to identify; defaults to current working copy
+     */
+    public @CheckForNull String branch(FilePath repository, @Nullable String rev) throws IOException, InterruptedException {
+        String branch = popen(repository, listener, false, new ArgumentListBuilder("id", "--branch", rev == null ? null : "--rev", rev == null ? null : rev)).trim();
```

Comment:
We use a bookmark that will be moved to revisions on different branches to release that state. The jenkins job is configured to checkout that bookmark. So we don't have any information on which branch the bookmark currently is set.